pandas/core/sorting.py

```diff
@@ -285,6 +316,24 @@ def nargsort(
     return indexer
 
 
+def ensure_key_mapped(values, key):
+    from pandas import Index
+
+    if not key:
+        return values
+
+    if isinstance(values, Index):
+        return values.map(key, na_action="ignore")
+    elif isinstance(values, np.ndarray):
+
+        def map_f(values, f):
```

Comment:
Why do you need this closure? Can we call `lib.map_infer_mask` directly?

Reply:
Yes. Updated. The closure was just used to simplify the implementation in another internal function that used it.